src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -56,6 +57,80 @@ pub struct TargetInfo {
     pub rustdocflags: Rc<[String]>,
 }
 
+/// Check Config (aka `--print=check-cfg` representation)
+#[derive(Debug, Default, Clone)]
+pub struct CheckCfg {
+    /// Is --check-cfg activated
+    pub exhaustive: bool,
+    /// List of expected cfgs
+    pub expecteds: HashMap<String, ExpectedValues<String>>,
+}
+
+/// List of expected check-cfg values
+#[derive(Debug, Clone)]
+pub enum ExpectedValues<T> {
+    /// List of expected values
+    Some(HashSet<Option<T>>),
+    /// All values expected
+    Any,
+}
+
+impl CheckCfg {
+    fn process_line(&mut self, line: &str) {
```

Comment:
As this re-implements some rustc logic, are there any unit tests they have that would be worth pulling over to make sure we conform on what any of this means/

Reply:
`rustc` currently only has unit tests for `--check-cfg` to `--print=check-cfg`, but not the inverse.
[tests/run-make/print-check-cfg/rmake.rs](https://github.com/rust-lang/rust/blob/76ed7a1fa40c3f54d3fd3f834e12bf9c932d0146/tests/run-make/print-check-cfg/rmake.rs)